### PR TITLE
fix: Slot reservation in new booker

### PR DESCRIPTION
**Pull request by roae**
## What does this PR do?

Bring back the slot reservation sistem to the new booker

## Type of change

- Bug fix (non-breaking change which fixes an issue)

### Seated event
- [x] Open the new booker in an event with seats 
- [x] Select an available slot
- [x] Open the same event and date in other browser, the seat should be already reserved.
- [x] In the first browser come back to the slots views or close the booking form
- [x] In the second browser reload the booking page, the reserved slot should be realeased. 

### No seated event
- [x] Open the new booker
- [x] Select an available slot
- [x] Open the same event and date in other browser, the seat should be disabled

## Mandatory Tasks

- [x] Make sure you have self-reviewed the code. A decent size PR without self-review might be rejected.

### CODE REVIEW

File: packages/features/bookings/Booker/components/BookEventForm/BookEventForm.tsx [hunk 1]
```diff
@@ -55,6 +63,30 @@ export const BookEventForm = ({ onCancel }: BookEventFormProps) => {
...
+    reserveSlot();
+    const interval = setInterval(reserveSlot, parseInt(MINUTES_TO_BOOK) * 60 * 1000 - 2000);
+    return () => {
+      if (eventType) {
+        releaseSlotMutation.mutate();
+        clearInterval(interval);
+      }
+    };
+    // eslint-disable-next-line react-hooks/exhaustive-deps
+  }, [eventType]);
```
Comment (roae) [inline packages/features/bookings/Booker/components/BookEventForm/BookEventForm.tsx:88]: reserve the slot on the booking form mount, and release on unmount the component

**Review by joeauyeung — DISMISSED** [review]
@roae reserving slots should be disabled for seated events except for the last seat. Here's a link to what I've done. Could you address this in your PR?

https://github.com/calcom/cal.com/pull/9778/files#r1240959003

**Comment by roae** [comment]
> @roae reserving slots should be disabled for seated events except for the last seat. Here's a link to what I've done. Could you address this in your PR?
> 
> https://github.com/calcom/cal.com/pull/9778/files#r1240959003


Hey @joeauyeung I believe this is how it works, when select an available slot in an seated event the number of available seats should decrease
imagen

**Comment by joeauyeung** [comment]
> > @roae reserving slots should be disabled for seated events except for the last seat. Here's a link to what I've done. Could you address this in your PR?
> > https://github.com/calcom/cal.com/pull/9778/files#r1240959003
> 
> Hey @joeauyeung I believe this is how it works, when select an available slot in an seated event the number of available seats should decrease imagen

We found that the reserve system was creating bugs with seats when multiple people tried signing up. 

For example, a customer sent an email blast about an event. This caused a big spike in people trying to sign up for the booking, but the reserve system was blocking multiple people trying to book the event at the same time even though there were seats available. 

I think if there are multiple seats available we don't need to reserve that slot.

**Comment by roae** [comment]
> > > @roae reserving slots should be disabled for seated events except for the last seat. Here's a link to what I've done. Could you address this in your PR?
> > > https://github.com/calcom/cal.com/pull/9778/files#r1240959003
> > 
> > 
> > Hey @joeauyeung I believe this is how it works, when select an available slot in an seated event the number of available seats should decrease imagen
> 
> We found that the reserve system was creating bugs with seats when multiple people tried signing up.
> 
> For example, a customer sent an email blast about an event. This caused a big spike in people trying to sign up for the booking, but the reserve system was blocking multiple people trying to book the event at the same time even though there were seats available.
> 
> I think if there are multiple seats available we don't need to reserve that slot.

Ok!, I get it

**Comment by ericrommel** [comment]
@roae @joeauyeung I'm seeing a different text from what the screenshot in the description has. Could you please confirm which one is correct?
image

**Comment by roae** [comment]
> @roae @joeauyeung I'm seeing a different text from what the screenshot in the description has. Could you please confirm which one is correct? image

Hey @ericrommel both are correct, My screenshot show text in spanish

**Comment by ericrommel** [comment]
> > @roae @joeauyeung I'm seeing a different text from what the screenshot in the description has. Could you please confirm which one is correct? image
> 
> Hey @ericrommel both are correct, My screenshot show text in spanish

@roae, I'm talking about the numbers there. Your screenshot shows "2 seats available" and my one shows "2 / 3 seats available".

**Comment by roae** [comment]
> > > @roae @joeauyeung I'm seeing a different text from what the screenshot in the description has. Could you please confirm which one is correct? image
> > 
> > 
> > Hey @ericrommel both are correct, My screenshot show text in spanish
> 
> @roae, I'm talking about the numbers there. Your screenshot shows "2 seats available" and my one shows "2 / 3 seats available".

yes there are differences in the translation.
 could you ask in threads?, I think both are correct, lets disscuss this with @Jaibles

**Review by ericrommel — APPROVED** [review]
Everything looks great from my tests.

**Review by zomars — APPROVED** [review]
Code LGTM. Let's address seated events in a follow up.

[cool](https://app.graphite.dev/macros/calcom)

**Pull request merged**